services-toolkit/explanation/four-levels-of-service-consumption.hbs.md

```diff
@@ -0,0 +1,69 @@
+
+Apps teams can discover the available classes using the `tanzu service class list` command. Now, rather than creating a ResourceClaim, they instead create a higher-level abstraction - a ClassClaim. The ClassClaim simply refers to the name of a class, and that's it. There's no longer a need to provide detailed information such as the GVK for the service resource behind the class. Simply create a ClassClaim referring to a class and then bind your application workload to the ClassClaim.
+
+Taking this into consideration it's clear that at level 3 it is much, much simpler for apps teams to consume services and that the separation of concerns is much neater. Still, a few limitations remain - service resources must still be bindable and ops teams still need to manually provision the service resources in order to fill the pool. Enter level 4 ...
+
+## Level 4 - ClassClaims and Provisioner-based Classes (aka "Dynamic Provisioning")
+
+Level 4 is the current highest level of service consumption in TAP. Released in TAP 1.5, it brings with it the introduction of provisioner-based classes, which, together with ClassClaims, power TAP's dynamic provisioning capability.
+
+![Diagram shows level 4 of service consumption in TAP](../../images/stk-4-levels-4.png)
+
+We have the same setup as before, but this time round the class defines a provisioner rather than a pool. We currently support one provisioner type - [Crossplane](https://www.crossplane.io/), but it's possible that support for new provisioners may be added in the future.
+
+When a ClassClaim which refers to a provisioner-based class is created, the services toolkit controller will ask the provisioner to provision whatever resources may be necessary to create a service instance which can then be bound to application workloads. In the example here, the provisioner creates a namespace, a service resource and a secret which conforms to the binding specification. Then, that secret is wired all the way back through to the application workload.
+
+Two big advantages are realized at level 4. Firstly, ops teams no longer need to manually pre-provision service resources. They are now created on-demand as and when needed. This not only helps to remove unnecessary burden from ops teams, but also helps to provider better utilization of resources as services resources no longer need to sit around in a pool waiting to be claimed. The second huge advantage is that service resources no longer need to be bindable. The provisioner can act almost like an adapter to bring pretty much any service you can think of into TAP. The only requirement is that the provisioner create a binding-conformant secret holding credentials for the provisioned service resources, but this is something that can be configured once during the dynamic provisioning setup.
```

Comment:
**nit:** You mention the _"the need to manually pre-provision service resources"_. I would argue that level 3 doesn't force ops to necessarily PRE-provision resources. To the platform, it doesn't matter whether a resources is being created before or after the claim, or before or after the pooled class.

Reply:
Yep makes sense - I'll remove the `pre-` prefix.